crates/lib3h/src/rrdht_util/r_value.rs

```diff
@@ -0,0 +1,335 @@
+use crate::rrdht_util::{Arc, ARC_LENGTH_MAX, ARC_RADIUS_MAX};
+
+/// builder for our knowledge of a single agent's storage / reachability
+pub struct RValuePeerRecord {
+    storage_arc: Arc,
+    uptime_0_to_1: f64,
+}
+
+impl Default for RValuePeerRecord {
+    fn default() -> Self {
+        Self {
+            storage_arc: Arc::new(0.into(), 0),
+            uptime_0_to_1: 0.0,
+        }
+    }
+}
+
+impl RValuePeerRecord {
+    /// what arc is this agent claiming to store
+    pub fn storage_arc(mut self, storage_arc: Arc) -> Self {
+        self.storage_arc = storage_arc;
+        self
+    }
+
+    /// what is this agent's uptime
+    /// - 0 = never reachable
+    /// - 1 = always reachable
+    pub fn uptime_0_to_1(mut self, uptime_0_to_1: f64) -> Self {
+        self.uptime_0_to_1 = uptime_0_to_1;
+        self
+    }
+}
+
+/// builder for a set of peers we know about
+/// this should generally either be our entire storage arc of peer info
+/// or, if it is huge, small sample arc
+pub struct RValuePeerRecordSet {
+    arc_of_included_peer_records: Arc,
+    peer_records: Vec<RValuePeerRecord>,
+}
+
+impl Default for RValuePeerRecordSet {
+    fn default() -> Self {
+        Self {
+            arc_of_included_peer_records: Arc::new(0.into(), 0),
+            peer_records: Vec::new(),
+        }
+    }
+}
+
+impl RValuePeerRecordSet {
+    /// the arc we are representing with this PeerRecordSet
+    pub fn arc_of_included_peer_records(mut self, arc_of_included_peer_records: Arc) -> Self {
+        self.arc_of_included_peer_records = arc_of_included_peer_records;
+        self
+    }
+
+    /// push a new PeerRecord into this record set
+    pub fn push_peer_record(mut self, peer_record: RValuePeerRecord) -> Self {
+        if !self
+            .arc_of_included_peer_records
+            .contains_location(peer_record.storage_arc.center())
+        {
+            panic!("peer record does not fit within arc_of_included_peer_records");
```

Comment:
Suggest "peer record location is not located within arc_of_included_peer_records" instead. A "does not fit" error could be misinterpreted as the arc of of the peer record not being contained by the arc_of_included_peer_records. I sense this could lead to some needless & misled debugging efforts.